cmd/genji/insert.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/genjidb/genji"
+	"github.com/genjidb/genji/document"
+	"github.com/genjidb/genji/engine"
+	"github.com/genjidb/genji/engine/badgerengine"
+	"github.com/genjidb/genji/engine/boltengine"
+)
+
+func skipSpaces(r *bufio.Reader) (byte, error) {
+	var c byte
+	var err error
+
+	for {
+		c, err = r.ReadByte()
+		if err != nil {
+			return c, err
+		}
+
+		if c != '\n' && c != '\r' && c != ' ' && c != '\t' {
+			break
+		}
+	}
+
+	return c, nil
+}
+
+func executeInsertCommand(db *genji.DB, table string, r io.Reader) error {
+	q := fmt.Sprintf("INSERT INTO %s VALUES ?", table)
+	rd := bufio.NewReader(r)
+	var c byte
+	var err error
+
+	// Ignore spaces.
+	c, err = skipSpaces(rd)
+	if err != nil {
+		return err
+	}
+	switch c {
+	case '{': // Json objects
+		if err := rd.UnreadByte(); err != nil {
+			return err
+		}
+
+		dec := json.NewDecoder(rd)
+		// NewFromDecoder is a wrapper to call parseJsonDocument.
+		d, err := document.NewFromDecoder(dec)
+		if err != nil {
+			return err
+		}
+
+		if err := db.Exec(q, &d); err != nil {
+			return err
+		}
+	case '[': // Array of json objects
+		if err := rd.UnreadByte(); err != nil {
+			return err
+		}
+
+		dec := json.NewDecoder(rd)
+		// Unmarshall json returns []interface{} for JSON arrays.
+		var m []interface{}
+		for dec.More() {
+
+			err := dec.Decode(&m)
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return err
+			}
+		}
+
+		// Iterate over m []interface{} to get map[string]interface{} then call NewFromMap
+		// that requires map[string]interface{}.
+		for i := range m {
+			d, err := document.NewFromMap(m[i])
+			if err != nil {
+				return err
+			}
+
+			if err := db.Exec(q, &d); err != nil {
+				return err
+			}
+		}
+	default:
+		return fmt.Errorf("found %q, but expected '{' or '['", c)
+
+	}
+
+	return nil
+}
+
+func runInsertCommand(e, DBPath, table string, data []string) error {
+	var ng engine.Engine
+	var err error
+
+	switch e {
+	case "bolt":
+		ng, err = boltengine.NewEngine(DBPath, 0660, nil)
+	case "badger":
+		ng, err = badgerengine.NewEngine(badger.DefaultOptions(DBPath).WithLogger(nil))
+	}
+
```

Comment:
Always group related code together, blank lines should be used to divide logical groups
```suggestion
```